var/spack/repos/builtin/packages/py-hail/package.py

```diff
@@ -0,0 +1,234 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyHail(MakefilePackage):
+    """Cloud-native genomic dataframes and batch computing (Python API)"""
+
+    homepage = "https://hail.is"
+    git = "https://github.com/hail-is/hail.git"
+    # We can't use tarballs because HAIL needs to look up git commit metadata
+    # to determine its version. We could patch this, but that is not yet
+    # implemented.
+    # url = "https://github.com/hail-is/hail/archive/refs/tags/0.2.130.tar.gz"
+
+    maintainers("teaguesterling")
+    license("MIT", checked_by="teaguesterling")
+
+    version("0.2.132", commit="678e1f52b9999cb05ebf03fd360e5c4506bd6dad")
+    version("0.2.131", commit="11d9b2ff89da9ef6a4f576be89f1f06959580ea4")
+    version("0.2.130", commit="bea04d9c79b5ca739364e8c121132845475f617a")
+    version("0.2.129", commit="41126be2df04e4ef823cefea40fba4cadbe5db8a")
+
+    resource(
+        name="catch",
+        url="https://github.com/catchorg/Catch2/releases/download/v2.6.0/catch.hpp",
+        sha256="a86133b34d4721b6e1cf7171981ea469789f83f2475907b4033012577e4975fe",
+        destination="hail/src/main/resources/include/catch.hpp",
+        expand=False,
+    )
+
+    resource(
+        name="libsimdpp-2.1",
+        extension="tar.gz",
+        url="https://storage.googleapis.com/hail-common/libsimdpp-2.1.tar.gz",
+        sha256="b0e986b20bef77cd17004dd02db0c1ad9fab9c70d4e99594a9db1ee6a345be93",
+        destination="hail/src/main/c",
+    )
+
+    resource(
+        name="mill-0.11.7",
+        url="https://repo1.maven.org/maven2/com/lihaoyi/mill-dist/0.11.7/mill-dist-0.11.7.jar",
+        sha256="278b430150af899495d360d1f886e223e78bb4a20e67144a240bfb7e2d4f6085",
+        destination="hail/mill",
+        expand=False,
+    )
+
+    variant("native", default=True, description="Compile native HAIL backend")
```

Comment:
What does "native" mean in this context? If this is related to `-march=` kind of flags, can we avoid the variant and inspect the `spec.target`?

Reply:
It's a naming thing. There's a way to build the HAIL with some prepackaged shared libraries, IIRC. Maybe worth removing.

Reply:
Updated the description (and also refined to dependencies around this). Tested both builds.